tiled/src/main/kotlin/ktx/tiled/tiledMap.kt

```diff
@@ -0,0 +1,99 @@
+package ktx.tiled
+
+import com.badlogic.gdx.maps.MapLayer
+import com.badlogic.gdx.maps.MapObject
+import com.badlogic.gdx.maps.MapProperties
+import com.badlogic.gdx.maps.tiled.TiledMap
+
+private val DEFAULT_MAP_LAYER = MapLayer().apply { name = "ktx-default-map-layer" }
```

Comment:
I don't think this is a good approach, especially since `MapLayer` objects seem to be mutable.

Reply:
You are right about the mutable part. It could be changed from the outside which is not nice.

The reason why I do it like this is because of my use-cases that I had. E.g. when I load a map I parse different layers and do something with them:
```kotlin
// this refers to a TiledMap
createSceneryEntities(this)
createEnemyEntities(this)
createSavepoints(this)
createNPCs(this)
createItemEntities(this)
createPortalEntities(this)
createTriggers(this)
```

One of these methods then looks like this:
```kotlin
private fun createSceneryEntities(map: Map) {
    map.mapObjects(LAYER_COLLISION).forEach { mapObj ->
        // loop through all map objects of that layer and
        // create the scenery entity according to the shape
        // of the map object.
        ecsEngine.scenery(world, mapObj.shape)
    }
}
```

So I don't care if the layer is there or not. If it is there then I will iterate over its objects and create collision scenery box2d objects out of it.
I would not like to have an exception because it is totally valid to leave out the collision layer.
Also, I would not like to have `isLayerExisting` calls everywhere. Granted, with Kotlin it is just a simple `?` character but especially when it comes to chaining I might end up with several `?`.

Of course this was my use-case and we should think here of a more generic approach for everyone. Just wanted to explain to you why I do it that way.

At least for me we can get rid of it since the `forEachMapObject` extension can take care of it.

I will make the change.